sched/environ/env_foreach.c

```diff
@@ -65,22 +65,23 @@ int env_foreach(FAR struct task_group_s *group,
                 env_foreach_t cb,
                 FAR void *arg)
 {
-  FAR char *ptr;
-  FAR char *end;
   int ret = OK;
+  int i;
 
   /* Verify input parameters */
 
   DEBUGASSERT(group != NULL && cb != NULL);
 
-  /* Search for a name=value string with matching name */
+  if (group->tg_envp == NULL)
+    {
+      return ret;
+    }
 
-  end = &group->tg_envp[group->tg_envsize];
-  for (ptr = group->tg_envp; ptr < end; ptr += (strlen(ptr) + 1))
+  for (i = 0; group->tg_envp[i] != NULL; i++)
```

Comment:
Minor. Not even sure if should be applied:
```
for (i = 0; group->tg_envp != NULL && group->tg_envp[i] != NULL; i++)
```
this will eliminate
```
  if (group->tg_envp == NULL)
    {
      return ret;
    }
```
but will make a long line. Not sure what is better.

Reply:
I considered this form before, but prefer the current one.